pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala

```diff
@@ -157,13 +158,21 @@ private abstract class BaseRRDD[T: ClassTag, U: ClassTag](
             dataOut.writeInt(1)
           }
 
-          for (elem <- iter) {
-            if (parentSerialized) {
-              val elemArr = elem.asInstanceOf[Array[Byte]]
-              dataOut.writeInt(elemArr.length)
-              dataOut.write(elemArr, 0, elemArr.length)
-            } else {
-              printOut.println(elem)
+          if (parentSerialized == "rows") {
```

Comment:
This would mean that here we could write something like `parentSerialized == SerializationFormats.ROWS` instead of hard-coding strings. In general its better to keep hard coded strings in one file / class and then refer to those variables elsewhere. Makes it easier to extend or modify things.